addons/mail/static/src/components/message/message.xml

```diff
@@ -111,54 +111,110 @@
+                        <div 
+                            class="o_Message_bubbleWrap position-relative d-flex align-items-start"
+                            t-att-class="{
+                                'd-flex justify-content-end': messageView.isInChatWindowAndIsAlignedRight,
+                                'pe-5': messageView.isInChatWindowAndIsAlignedLeft and !messageView.composerViewInEditing,
+                                'ps-5': messageView.isInChatWindowAndIsAlignedRight and !messageView.composerViewInEditing,
+                                'pe-2': messageView.isInChatWindowAndIsAlignedLeft and messageView.composerViewInEditing,
+                                'ps-2': messageView.isInChatWindowAndIsAlignedRight and messageView.composerViewInEditing,
+                            }"
+                        >
+                            <div
+                                class="o_Message_bubble position-relative"
+                                t-att-class="{
+                                    'flex-grow-1': messageView.composerViewInEditing,
+                                    'me-5': messageView.isInChatWindowAndIsAlignedLeft and !messageView.composerViewInEditing,
+                                    'ms-5': messageView.isInChatWindowAndIsAlignedRight and !messageView.composerViewInEditing,
+                                    'p-3': messageView.message.prettyBody !== '' and !messageView.composerViewInEditing and messageView.message.isDiscussionOrNotification, 
+                                    'p-2': messageView.composerViewInEditing, 
+                                    'pb-2 pt-1': !messageView.message.isDiscussionOrNotification,
+                                }"
+                            >
+                                <div
+                                    t-if="messageView.message.prettyBody !== '' || messageView.composerViewInEditing"
+                                    class="o_Message_background position-absolute start-0 top-0 w-100 h-100"
+                                    t-att-class="{
+                                        'rounded-end-3': !(messageView.isInChatWindowAndIsAlignedRight),
+                                        'rounded-start-3': messageView.isInChatWindowAndIsAlignedRight,      
+                                        'rounded-bottom-3 border': messageView.message.isDiscussionOrNotification,
+                                        'rounded-3 opacity-50': !messageView.message.isDiscussionOrNotification,
+                                        'o-isAuthorNotCurrentUserOrGuest border-info bg-info-light': !messageView.message.isCurrentUserOrGuestAuthor and !messageView.message.isHighlighted and messageView.message.isDiscussionOrNotification, 
+                                        'border-success bg-success-light opacity-25': messageView.message.isCurrentUserOrGuestAuthor and !messageView.message.isHighlighted and messageView.message.isDiscussionOrNotification,
```

Comment:
Can group `o-isAuthorNotCurrentUserOrGuest` and `border-success bg-success-light opacity-25` together (same condition)

Reply:
Same as the previous one, `border-success ...` has `messageView.message.isCurrentUserOrGuestAuthor` in its condition while `o-isAuthorNotCurrentUserOrGuest` has the opposite.

Reply:
Indeed... ignore comment then

I have a hard time to see difference when there's `!` prefix on long expressions... I quite dislike the use of this transitive way of expression opposite of `x.y.z`, why language is not just `x.y.!z`😢